tests/sentry/integrations/utils/test_lifecycle_metrics.py

```diff
@@ -474,3 +474,207 @@ def test_default_sample_log_rate_is_one(
+
+        metric_obj = self.TestLifecycleMetric()
+
+        @with_event_lifecycle(metric_obj)
+        def test_function():
+            raise ExampleException("test error")
+
+        with pytest.raises(ExampleException):
+            test_function()
+
+        self._check_metrics_call_args(mock_metrics, "failure")
+
+    @mock.patch("sentry.integrations.utils.metrics.logger")
+    @mock.patch("sentry.integrations.utils.metrics.metrics")
+    def test_decorator_halt_on_specific_exceptions(
+        self, mock_metrics: mock.MagicMock, mock_logger: mock.MagicMock
+    ) -> None:
+        """Test that decorator records halt for specified exception types"""
+        from sentry.integrations.utils.metrics import with_event_lifecycle
+
+        metric_obj = self.TestLifecycleMetric()
+
+        @with_event_lifecycle(metric_obj, halt_on_exceptions=[ExampleException])
+        def test_function():
+            raise ExampleException("this should be a halt")
+
+        with pytest.raises(ExampleException):
+            test_function()
+
+        self._check_metrics_call_args(mock_metrics, "halted")
+        mock_logger.info.assert_called_once_with(
+            "integrations.slo.halted",
+            extra={
+                "integration_domain": "messaging",
+                "integration_name": "my_integration",
+                "interaction_type": "my_interaction",
+                "exception_summary": repr(ExampleException("this should be a halt")),
+            },
+        )
+
+    @mock.patch("sentry.integrations.utils.metrics.logger")
+    @mock.patch("sentry.integrations.utils.metrics.metrics")
+    def test_decorator_halt_vs_failure_exception_types(
+        self, mock_metrics: mock.MagicMock, mock_logger: mock.MagicMock
+    ) -> None:
+        """Test that decorator correctly distinguishes between halt and failure exceptions"""
+        from sentry.integrations.utils.metrics import with_event_lifecycle
+
+        class HaltException(Exception):
+            pass
+
+        class FailureException(Exception):
+            pass
+
+        metric_obj = self.TestLifecycleMetric()
+
+        @with_event_lifecycle(metric_obj, halt_on_exceptions=[HaltException])
+        def test_function_halt():
+            raise HaltException("should halt")
+
+        @with_event_lifecycle(metric_obj, halt_on_exceptions=[HaltException])
+        def test_function_failure():
+            raise FailureException("should fail")
+
+        # Test halt exception
+        with pytest.raises(HaltException):
+            test_function_halt()
+
+        self._check_metrics_call_args(mock_metrics, "halted")
+        mock_logger.info.assert_called_once()
+
+        # Reset mocks
+        mock_metrics.reset_mock()
+        mock_logger.reset_mock()
+
+        # Test failure exception
+        with pytest.raises(FailureException):
+            test_function_failure()
+
+        self._check_metrics_call_args(mock_metrics, "failure")
```

Comment:
instead of checking call args should we be using the [assert_failure_metric](https://github.com/getsentry/sentry/blob/master/src/sentry/testutils/asserts.py#L95)/assert_halt_metric funcs?